lms/djangoapps/bulk_email/tasks.py

```diff
@@ -568,81 +560,51 @@ def _send_course_email(entry_id, email_id, to_list, global_email_context, subtas
                 total_recipients_failed += 1
-                log.error(
-                    "BulkEmail ==> Status: Failed(SMTPDataError), Task: %s, SubTask: %s, EmailId: %s, \
-                    Recipient num: %s/%s, Email address: %s",
-                    parent_task_id,
-                    task_id,
-                    email_id,
-                    recipient_num,
-                    total_recipients,
-                    email
+                log.exception(
+                    f"BulkEmail ==> Status: Failed(SMTPDataError), Task: {parent_task_id}, SubTask: {task_id}, "
+                    f"EmailId: {email_id}, Recipient num: {recipient_num}/{total_recipients}, Recipient UserId: "
+                    f"{current_recipient['pk']}"
                 )
                 if exc.smtp_code >= 400 and exc.smtp_code < 500:  # lint-amnesty, pylint: disable=no-else-raise
                     # This will cause the outer handler to catch the exception and retry the entire task.
                     raise exc
                 else:
                     # This will fall through and not retry the message.
                     log.warning(
-                        'BulkEmail ==> Task: %s, SubTask: %s, EmailId: %s, Recipient num: %s/%s, \
-                        Email not delivered to %s due to error %s',
-                        parent_task_id,
-                        task_id,
-                        email_id,
-                        recipient_num,
-                        total_recipients,
-                        email,
-                        exc.smtp_error
+                        f"BulkEmail ==> Task: {parent_task_id}, SubTask: {task_id}, EmailId: {email_id}, Recipient "
+                        f"num: {recipient_num}/{total_recipients}, Email not delievered to user "
+                        f"{current_recipient['pk']} due to error: {exc.smtp_error}"
                     )
                     subtask_status.increment(failed=1)
 
             except SINGLE_EMAIL_FAILURE_ERRORS as exc:
                 # This will fall through and not retry the message.
                 total_recipients_failed += 1
-                log.error(
-                    "BulkEmail ==> Status: Failed(SINGLE_EMAIL_FAILURE_ERRORS), Task: %s, SubTask: %s, \
-                    EmailId: %s, Recipient num: %s/%s, Email address: %s, Exception: %s",
-                    parent_task_id,
-                    task_id,
-                    email_id,
-                    recipient_num,
-                    total_recipients,
-                    email,
-                    exc
+                log.exception(
+                    f"BulkEmail ==> Status: Failed(SINGLE_EMAIL_FAILURE_ERRORS), Task: {parent_task_id}, SubTask: "
+                    f"{task_id}, EmailId: {email_id}, Recipient num: {recipient_num}/{total_recipients}, Recipient "
+                    f"UserId: {current_recipient['pk']}"
```

Comment:
We're dropping the exception logging here. Not sure if that was on purpose.

Reply:
I changed this from a `log.error()` to a `log.exception()` which I thought automatically included the exception stacktrace. I didn't think I needed to include the exception anymore.

Reply:
Ohp! Missed that. You're right.